src/Apps/Order/Routes/NewPayment/index.tsx

```diff
@@ -0,0 +1,496 @@
+                      loading={isCommittingMutation}
+                    />
+                  </>
+                </Media>
+              </Flex>
+            }
+          />
+        </HorizontalPadding>
+      </>
+    )
+  }
+
+  private getStripeBillingAddress(formAddress: Address): stripe.TokenOptions {
+    const selectedBillingAddress = (this.needsAddress()
+      ? this.state.address
+      : this.props.order.requestedFulfillment) as Address
+    const {
+      name,
+      addressLine1,
+      addressLine2,
+      city,
+      region,
+      postalCode,
+      country,
+    } = selectedBillingAddress
+    return {
+      name,
+      address_line1: addressLine1,
+      address_line2: addressLine2,
+      address_city: city,
+      address_state: region,
+      address_zip: postalCode,
+      address_country: country,
+    }
+  }
+
+  private createCreditCard({ token, oneTimeUse }) {
+    commitMutation<NewPaymentRouteCreateCreditCardMutation>(
+      this.props.relay.environment,
+      {
+        onCompleted: (data, errors) => {
+          const {
+            createCreditCard: { creditCardOrError },
+          } = data
+
+          if (creditCardOrError.creditCard) {
+            this.setOrderPayment({
+              creditCardId: creditCardOrError.creditCard.id,
+            })
+          } else {
+            if (errors) {
+              errors.forEach(this.onMutationError.bind(this))
+            } else {
+              const mutationError = creditCardOrError.mutationError
+              this.onMutationError(
+                new ErrorWithMetadata(mutationError.message, mutationError),
+                mutationError.detail
+              )
+            }
+          }
+        },
+        onError: this.onMutationError.bind(this),
+        mutation: graphql`
+          mutation NewPaymentRouteCreateCreditCardMutation(
+            $input: CreditCardInput!
+          ) {
+            createCreditCard(input: $input) {
+              creditCardOrError {
+                ... on CreditCardMutationSuccess {
+                  creditCard {
+                    id
+                  }
+                }
+                ... on CreditCardMutationFailure {
+                  mutationError {
+                    type
+                    message
+                    detail
+                  }
+                }
+              }
+            }
+          }
+        `,
+        variables: {
+          input: { token, oneTimeUse },
+        },
+      }
+    )
+  }
+
+  private setOrderPayment({ creditCardId }) {
+    commitMutation<NewPaymentRouteSetOrderPaymentMutation>(
+      this.props.relay.environment,
+      {
+        onCompleted: (data, errors) => {
+          this.setState({ isCommittingMutation: false })
+
+          const {
+            ecommerceSetOrderPayment: { orderOrError },
+          } = data
+
+          if (orderOrError.order) {
+            this.props.router.push(`/orders/${this.props.order.id}/review`)
```

Comment:
We should route them to the status page here.

Reply:
done